src/main/java/com/robinpowered/sdk/RobinApi.java

```diff
@@ -134,6 +135,16 @@ public boolean hasCredential() {
     }
 
     /**
+     * Returns whether a the interface is authenticated with a valid credential.
+     * Valid credentials have non-null values.
+     *
+     * @return Whether a valid credential has been set.
+     */
+    public boolean isAuthenticated() {
+        return credential != null && credential.getValue() != null;
```

Comment:
Another thing maybe we force a non-empty value when instantiating the credential to begin with, so the value can't be `null/''`